include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -93,3 +93,52 @@ def MemoryOp : CalyxPrimitive<"memory", [
     )>
   ];
 }
+
+class CalyxLibraryOp<string mnemonic, list<OpTrait> traits = []> :
+  CalyxPrimitive<"lib_" # mnemonic, traits> {
+
+  let summary = "Defines an operation which maps to a Calyx library primitive";
+  let description = [{
+    This operation represents an instance of a Calyx library primitive.
+    A library primitive maps to some hardware-implemented component within the
+    native Calyx compiler.
+  }];
+
+  let arguments = (ins StrAttr:$instanceName);
+  let builders = [
+    OpBuilder<(ins "StringAttr":$instanceName, "::mlir::TypeRange":$resultTypes), [{
+      $_state.addAttribute("instanceName", instanceName);
+      $_state.addTypes(resultTypes);
+    }]>
+  ];
+}
+
+class BoolBinLibOp<string mnemonic> : CalyxLibraryOp<mnemonic, [
+    SameTypeConstraint<"left", "right">
+  ]> {
+  let results = (outs AnyType:$left, AnyType:$right, I1:$out);
+}
+
+def LtLibOp  : BoolBinLibOp<"lt"> {}
+def GtLibOp  : BoolBinLibOp<"gt"> {}
+def EqLibOp  : BoolBinLibOp<"eq"> {}
+def NeqLibOp : BoolBinLibOp<"neq"> {}
+def GeLibOp  : BoolBinLibOp<"ge"> {}
+def LeLibOp  : BoolBinLibOp<"le"> {}
+
+class ArithBinLibOp<string mnemonic> : CalyxLibraryOp<mnemonic, [
```

Comment:
What we want eventually for both the arithmetic and boolean binary operations is access to direct ports, and the necessary functions for the `CellInterface` (#1597).

E.g.
https://github.com/llvm/circt/blob/c227f5c3d3596e5d8fd504b47342be361c5511d0/include/circt/Dialect/Calyx/CalyxPrimitives.td#L46-L60

Reply:
CellInterface is in progress! I should have a PR today.